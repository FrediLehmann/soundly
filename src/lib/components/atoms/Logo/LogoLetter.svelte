<script type="ts" context="module">
  import LogoLetterBlackPNG from './assets/logo_letter_black.png';
  import LogoLetterBlackWEBP from './assets/logo_letter_black.webp';
  import LogoLetterWhitePNG from './assets/logo_letter_white.png';
  import LogoLetterWhiteWEBP from './assets/logo_letter_white.webp';
  import LogoLetterPrimaryPNG from './assets/logo_letter_primary.png';
  import LogoLetterPrimaryWEBP from './assets/logo_letter_primary.webp';

  const images = {
    png: {
      black: LogoLetterBlackPNG,
      white: LogoLetterWhitePNG,
      primary: LogoLetterPrimaryPNG
    },
    webp: {
      black: LogoLetterBlackWEBP,
      white: LogoLetterWhiteWEBP,
      primary: LogoLetterPrimaryWEBP
    }
  };
</script>

<script type="ts">
  export let color: 'black' | 'white' | 'primary' = 'primary';
  export let size: 'small' | 'medium' | 'big' = 'small';
</script>

<picture data-size={size}>
  <source srcset={images.webp[color]} />
  <img alt="Noiceless Logo" src={images.png[color]} />
</picture>

<style>
  img {
    aspect-ratio: 1/1;
  }

  picture[data-size='small'] > img {
    height: 1.25rem;
  }

  picture[data-size='medium'] > img {
    height: 2.5rem;
  }

  picture[data-size='big'] > img {
    height: 4rem;
  }
</style>
