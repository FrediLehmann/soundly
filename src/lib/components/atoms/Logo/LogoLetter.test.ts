/**
 * @jest-environment jsdom
 */

import '@testing-library/jest-dom';
import { render } from '@testing-library/svelte';

import LogoLetter from './LogoLetter.svelte';

describe('<LogoLetter /> component', () => {
  test('renders correctly', () => {
    const { component } = render(LogoLetter);
    expect(component).toMatchSnapshot();
  });
});
