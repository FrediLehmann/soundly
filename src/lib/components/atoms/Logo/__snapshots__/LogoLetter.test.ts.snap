// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LogoLetter /> component renders correctly 1`] = `
LogoLetter {
  "$$": Object {
    "after_update": Array [],
    "before_update": Array [],
    "bound": Object {},
    "callbacks": Object {},
    "context": Map {},
    "ctx": Array [
      "primary",
      "small",
    ],
    "dirty": Array [
      -1,
    ],
    "fragment": Object {
      "c": [Function],
      "d": [Function],
      "i": [Function],
      "l": [Function],
      "m": [Function],
      "o": [Function],
      "p": [Function],
    },
    "not_equal": [Function],
    "on_destroy": Array [
      [Function],
    ],
    "on_disconnect": Array [],
    "on_mount": Array [],
    "props": Object {
      "color": 0,
      "size": 1,
    },
    "root": <div>
      <picture
        class="svelte-37lfet"
        data-size="small"
      >
        <source />
         
        <img
          alt="Noiceless Logo"
          class="svelte-37lfet"
        />
      </picture>
    </div>,
    "skip_bound": false,
    "update": [Function],
  },
  "$$set": [Function],
  "$capture_state": [Function],
  "$inject_state": [Function],
}
`;
